packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -62,138 +95,208 @@ function WorklistDataGrid(props) {
-    elementWithTabStop.focus();
-    event.preventDefault();
+  const handleNavigationModeChange = (isGridNavigationEnabled) => {
+    setIsNavigationEnabled(isGridNavigationEnabled);
   };
 
-  const handleClick = (event) => {
-    // Determine which cell was clicked. In the event that the user holds the mouse across multiple cells,
-    // the originating cell is the clicked cell/active element.
-    const clickedCell = event.target.closest('td,th') || document.activeElement.closest('td,th');
-    if (!clickedCell) {
-      // If anything other than a table data or table header cell is clicked, ignore the click.
-      return;
+  const areRowsSelected = () => (
+    rows.find(r => r.isSelected === true)
+  );
+
+  const clearRowSelection = () => {
+    if (areRowsSelected()) {
+      // Esc (while in row selection mode and rows are selected): Clear selection
+      if (onClearSelectedRows) {
+        onClearSelectedRows();
+      }
+    } else if (onClearRowSelectionMode) {
+      onClearRowSelectionMode(false);
     }
-    // Remove Tab stop from previous cell in table that has focus and set it to the cell that was clicked.
-    grid.current.rows[focusedRow.current].cells[focusedCol.current].tabIndex = -1;
-    focusedRow.current = clickedCell.parentElement.rowIndex;
-    focusedCol.current = clickedCell.cellIndex;
-
-    clickedCell.tabIndex = 0;
-    clickedCell.focus();
-    event.preventDefault();
   };
 
-  const getCellData = (cell, cellColumnIndex) => {
-    const tabIndex = { tabIndex: '-1' };
-    return props.rowHeaderIndex === cellColumnIndex ? (<th key={cellColumnIndex} {...tabIndex} className={cx('worklist-data-grid-row-header')}>{cell.content}</th>) : (<td key={cellColumnIndex} {...tabIndex} className={cx('worklist-data-grid-cell-data')}>{cell.content}</td>);
+  const handleMoveCellFocus = (fromCell, toCell) => {
+    removeTabStop(fromCell.x, fromCell.y);
+    setFocusedRowCol(toCell.x, toCell.y);
+    setIsNavigationEnabled(true);
+  };
+
+  const handleCellSelectionChange = (rowId, columnId, cellCoordinates) => {
+    setFocusedRowCol(cellCoordinates.x, cellCoordinates.y);
+    setIsNavigationEnabled(true);
+
+    setCurrentSelectedCell((rowId && columnId) ? { rowId, columnId } : null);
   };
 
-  const buildColumn = (columnData) => {
-    const width = columnData.width || props.columnWidth;
+  const getRowSelectionCellData = (cellRowIndex, row) => (
+    <RowSelectionCell
+      rowId={row.id}
+      columnId={displayedColumns[0].id}
+      coordinates={{ x: cellRowIndex, y: 0 }}
+      acceptsFocus={focusedRow.current === cellRowIndex && focusedCol.current === 0}
+      isSelected={row.isSelected}
+      ariaLabel={row.ariaLabel}
+      onCellSelectionChange={handleCellSelectionChange}
+      onMoveCellFocus={handleMoveCellFocus}
+      isRowSelectionModeEnabled={hasSelectableRows}
+      isNavigationEnabled={isNavigationEnabled}
+      rowsLength={rows.length}
+      columnsLength={displayedColumns.length}
```

Comment:
The cell should not need most of these props.  The Row component will know about row selection mode.  The WorklistDataGrid can handle navigation as the parent container.